Core/Src/main.c

```diff
@@ -1,181 +1,224 @@
+#include <string.h>
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+
+#define MAX_ADC_VALUE 1024
+#define MIN_DUTY_CYCLE_COUNT 100 /* 5% of timer period */
+#define MAX_DUTY_CYCLE_COUNT 200 /* 10% of timer period */
+
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+  uint8_t start_byte = 1;
+  uint8_t spi_recv_buf[2] = {0};
+  uint16_t adc_value = 0;
+
+  float pwm_scale = 0.0f;
+  uint32_t pwm_offset = 0;
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  /* USER CODE BEGIN 2 */
+
+  /* set CS to be high by default */
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+  /* start timer */
+  TIM1->CCR1 = MIN_DUTY_CYCLE_COUNT;
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1)
+  {
+	/* 1. Read potentiometer value from ADC */
+    /* pull CS to low */
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+    /* send start byte to ADC */
+    HAL_SPI_Transmit(&hspi1, &start_byte, 1, 100);
+    /* receive 2 bytes from ADC */
+    HAL_SPI_Receive(&hspi1, spi_recv_buf, 2, 100);
+    /* pull CS back to high */
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+    /* store result from ADC in a 16-bit number */
+    adc_value = (spi_recv_buf[0] << 8) + spi_recv_buf[1];
+    /* clear the 6 MSB as they're don't-cares */
+    adc_value &= 0x03FF;
+
+	/* 2. Calculate motor PWM output */
+    pwm_scale = (float)adc_value / MAX_ADC_VALUE;
+    pwm_offset = (uint32_t)((MAX_DUTY_CYCLE_COUNT - MIN_DUTY_CYCLE_COUNT) * pwm_scale);
+    TIM1->CCR1 = MIN_DUTY_CYCLE_COUNT + pwm_offset;

```

Comment:
TIM1->CCR1 works fine but there is also a HAL macro that does it for you called __HAL_TIM_SET_COMPARE that can be found in the tim hal drivers h file. Your choice on which one you want to use.

Reply:
I switched to using __HAL_TIM_SET_COMPARE. I like macros :)